iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -196,73 +230,129 @@ private static void writeExtensions(Network n, NetworkXmlWriterContext context)
                 }
+                writeEndElement(writer);
             }
-            for (HvdcLine l : n.getHvdcLines()) {
-                if (!filter.test(l.getConverterStation1()) || !filter.test(l.getConverterStation2())) {
-                    continue;
-                }
-                HvdcLineXml.INSTANCE.write(l, n, context);
+        }
+    }
+
+    public static void writeBaseNetwork(Network n, NetworkXmlWriterContext context, BusFilter filter) throws XMLStreamException {
+
+        for (Substation s : n.getSubstations()) {
+            SubstationXml.INSTANCE.write(s, null, context);
+        }
+        for (Line l : n.getLines()) {
+            if (!filter.test(l)) {
+                continue;
+            }
+            if (l.isTieLine()) {
+                TieLineXml.INSTANCE.write((TieLine) l, n, context);
+            } else {
+                LineXml.INSTANCE.write(l, n, context);
+            }
+        }
+        for (HvdcLine l : n.getHvdcLines()) {
+            if (!filter.test(l.getConverterStation1()) || !filter.test(l.getConverterStation2())) {
+                continue;
             }
+            HvdcLineXml.INSTANCE.write(l, n, context);
+        }
+    }
 
-            if (!options.isSkipExtensions()) {
+    public static Anonymizer write(Network n, ExportOptions options, OutputStream osb, OutputStream ose, DataSource dataSource) {
+        try {
+            final XMLStreamWriter writer = initializeWriter(n, osb, options);
+
+            BusFilter filter = BusFilter.create(n, options);
+            Anonymizer anonymizer = options.isAnonymized() ? new SimpleAnonymizer() : null;
+            NetworkXmlWriterContext context = new NetworkXmlWriterContext(anonymizer, writer, options, filter);
+
+            writeBaseNetwork(n, context, filter);
+
+            if (!options.isSkipExtensions() && !getNetworkExtensions(n).isEmpty()) {
+                final XMLStreamWriter extensionsWriter = initializeWriter(n, ose, options);
+                context.setExtensionsWriter(extensionsWriter);
                 // Consider the network has been exported so its extensions will be written also
                 context.addExportedEquipment(n);
-
                 // write extensions
-                writeExtensions(n, context);
+                if (options.isOneFilePerExtensionType()) {
+                    writeExtensionsInMultipleFile(n, context, dataSource, options);
+                } else {
+                    writeExtensions(n, context);
+                    writeEndElement(extensionsWriter);
+                }
             }
-
-            writer.writeEndElement();
-            writer.writeEndDocument();
+            writeEndElement(writer);
 
             return anonymizer;
-        } catch (XMLStreamException e) {
-            throw new UncheckedXmlStreamException(e);
+        } catch (XMLStreamException | IOException e) {
+            throw new UncheckedXmlStreamException((XMLStreamException) e);
```

Comment:
strange to cast as `XMLStreamException` after accepting both `XMLStreamException` and `IOException` ?